app/src/main/java/ch/admin/bag/dp3t/inform/InformViewModel.kt

```diff
@@ -106,7 +108,7 @@ class InformViewModel(application: Application, private val state: SavedStateHan
 		emit(Resource.loading(data = null))
 		val onsetResponseTime = onsetResponseTime ?: System.currentTimeMillis()
 		val timeBetweenOnsetAndUploadRequest = (System.currentTimeMillis() - onsetResponseTime).toInt()
-		delay(UPLOAD_REQUEST_TIME_PADDING - (System.currentTimeMillis() - onsetResponseTime) % UPLOAD_REQUEST_TIME_PADDING)
+		delay((SecureRandom().nextDouble() * UPLOAD_REQUEST_TIME_PADDING).roundToLong())
```

Comment:
Why do we need this random delay?

Reply:
To prevent having certain durations between onset and upload request with higher probability for real than for fake requests. with this additional random delay we smoothen the distribution of the real user interaction durations.